stdlib/public/SDK/Dispatch/Time.swift

```diff
@@ -53,17 +53,26 @@ public struct DispatchTime : Comparable {
 	///               system sleep time), not zero nanoseconds since boot.
 	public init(uptimeNanoseconds: UInt64) {
 		var rawValue = uptimeNanoseconds
-		if (DispatchTime.timebaseInfo.numer != DispatchTime.timebaseInfo.denom) {
-			rawValue = (rawValue * UInt64(DispatchTime.timebaseInfo.denom) 
-				+ UInt64(DispatchTime.timebaseInfo.numer - 1)) / UInt64(DispatchTime.timebaseInfo.numer)
+
+		// UInt64.max means distantFuture. Do not try to scale it.
+		if rawValue != UInt64.max && DispatchTime.timebaseInfo.numer != DispatchTime.timebaseInfo.denom {
```

Comment:
It seems like the scaling could always trigger an overflow, right? Is this sufficient?

Reply:
I protected the operations with overflow checks, apart from the division which can't overflow, so I don't think anything else can trap.

Reply:
So what is the `if` for? Is it simply trying to optimize some values on some platforms?

Reply:
The first part lets UInt64.max get through unchanged, the second part avoids the arithmetic when there is no timebase scaling. So yes, that part is an optimization (that was already there).